tests/scikit_learn/etl/test_split.py

```diff
@@ -1,24 +1,178 @@
+                           'petallength', 'petalwidth'], slice_size)]
+
+    arguments = {
+        'parameters': {},
+        'named_inputs': {
+            'input data': df[0],
+        },
+        'named_outputs': {
+            'split 1': 'split_1_task_1',
+            'split 2': 'split_2_task_1'
+        }
+    }
+    instance = SplitOperation(**arguments)
+    result = util.execute(instance.generate_code(),
+                          {'df': df[1]})
+
+    assert len(result['split_2_task_1']) == slice_size
+
+
+def test_split_fail_by_seed():
+    slice_size = 10
+    df = ['df', util.iris(['sepallength', 'sepalwidth',
+                           'petallength', 'petalwidth'], slice_size)]
+
+    arguments = {
+        'parameters': {'seed': 4294967296  # Seeds higher or equal to 4294967296, and lower than 0 break the code
+                       },
+        'named_inputs': {
+            'input data': df[0],
+        },
+        'named_outputs': {
+        }
+    }
+    instance = SplitOperation(**arguments)
+    with pytest.raises(ValueError) as InvalidSeed:
+        result = util.execute(instance.generate_code(),
+                              {'df': df[1]})
+    print(InvalidSeed)
+
+
+def test_weird_split_by_weights_and_slice():
+    """
+    This happens when the 'slice_size' is between 1 and 10
+    and the 'weights' value is between -9 and 9
+    I don't know if it's a bug or not
+    """
+    slice_size = 10
+    df = ['df', util.iris(['sepallength', 'sepalwidth',
+                           'petallength', 'petalwidth'], slice_size)]
+
+    arguments = {
+        'parameters': {'weights': 9},
+        'named_inputs': {
+            'input data': df[0],
+        },
+        'named_outputs': {
+        }
+    }
+    instance = SplitOperation(**arguments)
+    result = util.execute(instance.generate_code(),
+                          {'df': df[1]})
+    with pytest.raises(AssertionError) as Weird:
+        assert len(result['split_1_task_1']) and len(result['split_2_task_1']) != slice_size
+    print(Weird)
+
+
+def test_split_success_no_output_implies_no_code():
+    """
+    Change SplitOperation: it shouldn't generate
+    code if there is not output!
+    Does this happen to every operation???
+    """
+    slice_size = 10
+    df = ['df', util.iris(['sepallength', 'sepalwidth',
+                           'petallength', 'petalwidth'], slice_size)]
 
     arguments = {
         'parameters': {},
         'named_inputs': {
             'input data': df[0],
         },
         'named_outputs': {
-            'output data': 'out'
         }
     }
     instance = SplitOperation(**arguments)
-    result = util.execute(instance.generate_code(), 
-                          dict([df]))
-    assert result['out'].equals(util.iris(size=slice_size))
+    result = util.execute(instance.generate_code(),
+                          {'df': df[1]})
+    with pytest.raises(AssertionError) as HasCode:
```

Comment:
If it is raising an error even though it does not produce code, you can change the SplitOperation and probably mode the `has_validation` assignment to one of the first lines of the constructor and before raising the exception, check if `has_code` is `True`.

Reply:
Changed: `self.has_code = len(named_inputs) == 1` 
To: `self.has_code = len(named_outputs) and len(named_inputs) == 1` 
This way it won't generate any code without input or output.

Reply:
Maybe, for clarity:

```
self.has_code = len(named_outputs) >= 1 and len(named_inputs) == 1
```

Reply:
Ok